spec/components/guardian_spec.rb

```diff
@@ -2692,6 +2692,35 @@
     end
   end
 
+  describe 'can_use_flair_group?' do
+    fab!(:group) { Fabricate(:group, title: 'Groupie', flair_icon: 'icon') }
+
+    it 'is false without a logged in user' do
+      expect(Guardian.new(nil).can_use_flair_group?(user)).to be_falsey
+    end
+
+    it 'is false if the group does not exist' do
+      user.update(groups: [group])
+      expect(Guardian.new(user).can_use_flair_group?(user, nil)).to be_falsey
+      expect(Guardian.new(user).can_use_flair_group?(user, Group.last.id + 1)).to be_falsey
+    end
+
+    it 'is false if the user is not a part of the group' do
+      user.update(groups: [])
```

Comment:
I don't think this line is required too since we don't leak state between tests.